src/components/Dashboard/PendingTxs/PendingTxsList/index.tsx

```diff
@@ -0,0 +1,80 @@
+import { ReactElement, useMemo } from 'react'
+import { List } from '@material-ui/core'
+import styled from 'styled-components'
+import { useSelector } from 'react-redux'
+import Skeleton from '@material-ui/lab/Skeleton/Skeleton'
+
+import { Transaction } from 'src/logic/safe/store/models/types/gateway.d'
+import { sm } from 'src/theme/variables'
+import { currentChainId } from 'src/logic/config/store/selectors'
+import { generateSafeRoute, SAFE_ROUTES } from 'src/routes/routes'
+import { getChainById } from 'src/config'
+import PendingTxListItem from 'src/components/Dashboard/PendingTxs/PendingTxListItem'
+import { currentSafe } from 'src/logic/safe/store/selectors'
+import { useQueueTransactions } from 'src/routes/safe/components/Transactions/TxList/hooks/useQueueTransactions'
+
+const SkeletonWrapper = styled.div`
+  margin: ${sm} auto;
+  border-radius: 8px;
+  overflow: hidden;
+`
+
+const PendingTxsList = ({ size }: { size: number }): ReactElement | null => {
+  const { address } = useSelector(currentSafe)
+  const chainId = useSelector(currentChainId)
+
+  const queue = useQueueTransactions()
+
+  const queuedTxsToDisplay: Transaction[] = useMemo(() => {
+    let txs: Transaction[] = []
+
+    if (!queue) {
+      return txs
+    }
+
+    nonceLoop: for (const { transactions } of Object.values(queue)) {
+      for (const [, transactionsByNonce] of transactions) {
+        // Add same nonced transactions to list
+        txs = txs.concat(transactionsByNonce)
+
+        // If adding same nonced transactions exceeded/reached limit, cleanup and break
+        if (txs.length >= size) {
+          if (txs.length > size) {
+            txs = txs.slice(0, size)
+          }
+          break nonceLoop
+        }
+      }
+    }
```

Comment:
I would just show the first tx from each group:

```suggestion
const flatTxList = Object.values(queue).map(txns => txns[0]).slice(0, size)
```

Reply:
Can't that be misleading? Meaning if there are 2 transactions with the same nonce and we only show one of them, supressing the other. Thinking of the case of a rejection, I find value in displaying which transaction the rejection applies to.

Reply:
What's the point showing both an original tx and its rejection in that widget?

Reply:
I am not sure about selecting an opinionated one of a group. Take this scenario for example: a Safe controlled by more than two people with a group of three transactions. Not everyone will be opting for the first transaction.

Perhaps if we chose the one with the highest number of confirmations?